src/Microsoft.Owin.Security.Authorization/AuthorizationDependenciesProvider.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using Microsoft.Owin.Logging;
+using Microsoft.Owin.Security.Authorization.Infrastructure;
+
+namespace Microsoft.Owin.Security.Authorization
+{
+    public class AuthorizationDependenciesProvider
+    {
+        public AuthorizationDependenciesProvider(IAuthorizationPolicyProvider policyProvider = null,
+            IAuthorizationHandler[] handlers = null, ILoggerFactory loggerFactory = null)
+        {
+            OnCreate = options =>
+            {
+                policyProvider = policyProvider ?? new DefaultAuthorizationPolicyProvider(options);
+                handlers = handlers ?? new IAuthorizationHandler[] {new PassThroughAuthorizationHandler()};
+                loggerFactory = loggerFactory ?? new DiagnosticsLoggerFactory();
+                return new AuthorizationDependencies
+                {
+                    PolicyProvider = policyProvider,
+                    Service = new DefaultAuthorizationService(
+                        policyProvider,
+                        handlers,
+                        loggerFactory.Create("ResourceAuthorization"))
+                };
+            };
+            OnDispose = (options, context) => { };
+        }
+
+        public Func<AuthorizationOptions, AuthorizationDependencies> OnCreate { get; set; }
+
+        public Action<AuthorizationOptions, AuthorizationDependencies> OnDispose { get; set; }
```

Comment:
Im further puzzled why there is a departure from the standard Dispose pattern that has been implemented since .NET 1.0? What does this implementation solve that the existing pattern does not? What is the object lifetime flow for this pattern? How does the finalizer handle/interact with the object referenced by this lambda?

See https://msdn.microsoft.com/en-us/library/b1yfkh5e(v=vs.110).aspx